### PR TITLE
Add html input validation in the form component

### PR DESCRIPTION
**Pull request by ctkochan22**
### Details
WAF input rules filter out html/xml tags in most API parameters. Given that we rarely want users to submit html tags as inputs, we have decided to validate all inputs for html. And we will do so in the `Form` component.

Here is what it would look like:
_First/Last Name-_
image

_Workspage Settings -> General -_
image

### Fixed Issues
$ https://github.com/Expensify/App/issues/14610
PROPOSAL: [GH](https://github.com/Expensify/App/issues/14610#issuecomment-1433947933) - [Slack](https://expensify.slack.com/archives/C01GTK53T8Q/p1676596415700269)

### Tests

- [ ] Verify that no errors appear in the JS console

### QA Steps
Let's test that it picks up for html tags, but the focus here is to make sure we don't have regressions
- [ ] WorkspaceNewRoomPage.js
    - Create a new workspace room. 
    - Add a name with html tags `test<b>`. Verify that there is an error
    - Now delete part of it so that it looks like `test<` and verify that you get an error saying you can only have characters
    - Now delete it so that its just text `test`. Verify that error is gone. 
    - Select a workspace below that field. And submit and verify that you can create a workspace room
    - https://recordit.co/JgPnzb7FZf
- [ ] CloseAccountPage.js
    - Have an account you're willing to close
    - Go to Security -> Close Account
    - Type in a message with html tags `hello <b>`. Verify you get an error
    - Now remove the html tag, `hello`
    - Add the email and close the account. 
    - https://recordit.co/64J5ZbsT7i
- [ ] WorkspaceSettingsPage.js
    - Go to Workspace Settings -> General Settings. Input `<b> </b>` as the workspace name. Verify that you see an error when you blur the field or submit the form
    - Remove the tags, add a valid name, and confirm that workspace name saves successfully
    - https://recordit.co/WW3Ne7UKHZ
- [ ] DisplayName
    - Go to settings -> profile. Display Name
    - Add a name with a html tag in it `Kosuke<b>`. Verify that you get an error message
    - Remove the tag and verify that you can add a name successfully

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android / native
    - [x] Android / Chrome
    - [x] iOS / native
    - [x] iOS / Safari
    - [x] MacOS / Chrome / Safari
    - [x] MacOS / Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is correct English and approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [x] I have checked off every checkbox in the PR author checklist, including those that don't apply to this PR.

### Screenshots/Videos

0eTSnGSP6r

<details>
<summary>Web</summary>

https://recordit.co/0eTSnGSP6r

</details>

<details>
<summary>Mobile Web - Chrome</summary>

https://recordit.co/cIOjb7gHc7

</details>

<details>
<summary>Mobile Web - Safari</summary>

https://recordit.co/0L0CmlQRJF

</details>

<details>
<summary>Desktop</summary>

https://recordit.co/rrvronUa7p

</details>

<details>
<summary>iOS</summary>

https://recordit.co/6mxyX6mqLe

</details>

<details>
<summary>Android</summary>

https://recordit.co/F8zbIrxmPn

</details>

### CODE REVIEW

**Comment by ctkochan22** [comment]
Waiting on copy, but ready for an initial review. Been testing it pretty heavily

**Comment by eVoloshchak** [comment]
Tests well on all platforms
I think there are cases where this error superseding other errors isn't the best option, for instance on `BankAccountManualStep`. The error will be `Please enter a valid routing number` for every symbol or invalid combination of symbols, except for when there is a tag.
image
This is a small issue though, it will look better with `Invalid character` error

**Comment by ctkochan22** [comment]
Yeah that is a good catch @eVoloshchak . I'd say its a rare case in general. If someone putting anything resembling tags where a routing number is supposed to be, they should be notified that that act is not allowed. 

The "Please enter valid Routing Number" to me, is a "friendly" message for "friendly" users

**Comment by ctkochan22** [comment]
Copy updated. Ready for review

File: src/components/Form.js
Comment (tgolen) [inline src/components/Form.js:181]: The `CONST.js` file has `VALIDATE_CODE_REGEX_STRING`. Could you please move this regex to the same place and define it there?

File: src/components/Form.js
Comment (tgolen) [inline src/components/Form.js:185]: Let's rename this to `common.error.invalidCharacter` so that it matches the text

**Review by tgolen — DISMISSED** [review]

**Comment by ctkochan22** [comment]
Updated

**Comment by eVoloshchak** [comment]
## Reviewer Checklist

- [ ] I have verified the author checklist is complete (all boxes are checked off).
- [ ] I verified the correct issue is linked in the `### Fixed Issues` section above
- [ ] I verified testing steps are clear and they cover the changes made in this PR
    - [ ] I verified the steps for local testing are in the `Tests` section
    - [ ] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [ ] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [ ] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [ ] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I verified tests pass on **all platforms** & I tested again on:
    - [ ] Android / native
    - [ ] Android / Chrome
    - [ ] iOS / native
    - [ ] iOS / Safari
    - [ ] MacOS / Chrome / Safari
    - [ ] MacOS / Desktop
- [ ] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [ ] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [ ] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [ ] I verified that comments were added to code that is not self explanatory
    - [ ] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [ ] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
    - [ ] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [ ] I verified any copy / text that was added to the app is correct English and approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [ ] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [ ] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [ ] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [ ] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [ ] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [ ] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [ ] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [ ] If a new component is created I verified that:
    - [ ] A similar component doesn't exist in the codebase
    - [ ] All props are defined accurately and each prop has a `/** comment above it */`
    - [ ] The file is named correctly
    - [ ] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [ ] The only data being stored in the state is data necessary for rendering and nothing else
    - [ ] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [ ] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [ ] All JSX used for rendering exists in the render method
    - [ ] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [ ] If any new file was added I verified that:
    - [ ] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [ ] If a new CSS style is added I verified that:
    - [ ] A similar style doesn't already exist
    - [ ] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [ ] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [ ] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [ ] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [ ] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [ ] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Web</summary>



</details>

<details>
<summary>Mobile Web - Chrome</summary>



</details>

<details>
<summary>Mobile Web - Safari</summary>



</details>

<details>
<summary>Desktop</summary>



</details>

<details>
<summary>iOS</summary>



</details>

<details>
<summary>Android</summary>



</details>

**Comment by eVoloshchak** [comment]
It seems like it's validating only the first input, on Legal Name page tags aren't allowed only for the first name
image

**Comment by ctkochan22** [comment]
@tgolen Hoping you can take this over or finding someone who can

**Comment by tgolen** [comment]
@eVoloshchak Could I get your help on this to try and dig into the code for why it's only validating the first input there?

**Comment by eVoloshchak** [comment]
@tgolen, sure thing, I'll look into this tomorrow as a first priority

File: src/components/Form.js
Comment (eVoloshchak) [inline src/components/Form.js:181]:
```suggestion
        _.each(values, (inputValue, inputID) => {
```

**Review by eVoloshchak — CHANGES_REQUESTED** [review]

**Comment by eVoloshchak** [comment]
@tgolen, I've found what's causing the issue, we're using [_.every](https://www.geeksforgeeks.org/lodash-_-every-method/) to iterate over the object with inputs
> The _.every method checks if the predicate returns true for all elements of collection and **iteration is stopped once the predicate returns falsely**
We should use _.each instead.
I've added a comment, but I can't push into this branch

**Comment by tgolen** [comment]
Ah, excellent! I can make the change. Though, it sounds like we are still figuring out if we want to do this or not in Slack. Thank you!

**Comment by tgolen** [comment]
@eVoloshchak OK, I've committed that change for `_.each()`. I think we should move forward with this. If it turns out we want to revert this at some point in the future, I think that's OK.

I'm going to merge `main` into this branch too just to make sure it's up to date and fine for merging.

**Comment by eVoloshchak** [comment]
## Reviewer Checklist

- [ ] I have verified the author checklist is complete (all boxes are checked off).
- [ ] I verified the correct issue is linked in the `### Fixed Issues` section above
- [ ] I verified testing steps are clear and they cover the changes made in this PR
    - [ ] I verified the steps for local testing are in the `Tests` section
    - [ ] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [ ] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [ ] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [ ] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I verified tests pass on **all platforms** & I tested again on:
    - [ ] Android / native
    - [ ] Android / Chrome
    - [ ] iOS / native
    - [ ] iOS / Safari
    - [ ] MacOS / Chrome / Safari
    - [ ] MacOS / Desktop
- [ ] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [ ] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [ ] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [ ] I verified that comments were added to code that is not self explanatory
    - [ ] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [ ] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
    - [ ] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [ ] I verified any copy / text that was added to the app is correct English and approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [ ] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [ ] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [ ] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [ ] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [ ] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [ ] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [ ] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [ ] If a new component is created I verified that:
    - [ ] A similar component doesn't exist in the codebase
    - [ ] All props are defined accurately and each prop has a `/** comment above it */`
    - [ ] The file is named correctly
    - [ ] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [ ] The only data being stored in the state is data necessary for rendering and nothing else
    - [ ] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [ ] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [ ] All JSX used for rendering exists in the render method
    - [ ] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [ ] If any new file was added I verified that:
    - [ ] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [ ] If a new CSS style is added I verified that:
    - [ ] A similar style doesn't already exist
    - [ ] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [ ] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [ ] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [ ] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [ ] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [ ] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Web</summary>



</details>

<details>
<summary>Mobile Web - Chrome</summary>



</details>

<details>
<summary>Mobile Web - Safari</summary>



</details>

<details>
<summary>Desktop</summary>



</details>

<details>
<summary>iOS</summary>



</details>

<details>
<summary>Android</summary>



</details>

**Review by eVoloshchak — APPROVED** [review]

**Comment by tgolen** [comment]
@eVoloshchak Looks like this just needs an approved review from you now. Are you able?

**Review by eVoloshchak — DISMISSED** [review]

**Comment by tgolen** [comment]
Thanks! @marcaaron Could you add a review, please?

File: src/CONST.js
Comment (marcaaron) [inline src/CONST.js:495]: **noting (non-blocking):** nobody is going to know what the "WAF inputrules" are except internal Expensify. Maybe that's fine? Or we could mention something about how the API strips out things that are invalid to contextualize this.

File: src/components/Form.js
Comment (marcaaron) [inline src/components/Form.js:180]:
```suggestion
        // Validate the input for html tags. It should supercede any other error
```

Superseed is what I use to get the grass on my lawn going!

File: src/components/Form.js
Comment (marcaaron) [inline src/components/Form.js:182]: I can't quite remember, but are `inputValue` always `{String}`? 

I don't think `.search()` works on numbers. If I'm right maybe we can do something like this instead...

```suggestion
            if (!inputValue || (_.isString(inputValue) && inputValue.search(CONST.VALIDATE_FOR_HTML_TAG_REGEX) === -1)) {
```

File: src/CONST.js
Comment (tgolen) [reply src/CONST.js:495]: I can update this comment, that's a good suggestion.

**Comment by tgolen** [comment]
Updated

File: src/components/Form.js
Comment (marcaaron) [inline src/components/Form.js:183]: Sorry, I think the logic I gave is wrong actually :ohnothing: we want to return early and avoid reporting the error when:

- there's no input value at all
- it's not a string
- it is a string that has no html

so 
```js
if (!inputValue || !_.isString(inputValue) || inputValue.search(CONST.VALIDATE_FOR_HTML_TAG_REGEX) === -1)
```
if we can pass non-strings with the current example they would fail to return early because

- it has a value
- it's not a string (`_.isString(inputValue) && inputValue.search(CONST.VALIDATE_FOR_HTML_TAG_REGEX) === -1)` short circuits to false)
- html detected falsely

Comment (tgolen) [reply src/components/Form.js:183]: Ah, makes sense. Good catch.

**Review by marcaaron — DISMISSED** [review]

File: src/components/Form.js
Comment (marcaaron) [inline src/components/Form.js:183]: missing the opening curly brace

Comment (tgolen) [reply src/components/Form.js:183]: Akward :blush

**Comment by tgolen** [comment]
Retested to be sure my curly braces are working

image

**Review by marcaaron — APPROVED** [review]

**Pull request merged**
